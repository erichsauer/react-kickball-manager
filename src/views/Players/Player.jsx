import { useEffect, useState } from 'react'
import { Link } from 'react-router-dom'
import Loading from '../../components/loading/Loading'
import { getPlayerById } from '../../services/players'

function Player({
  user,
  match: {
    params: { id },
  },
}) {
  const [player, setPlayer] = useState({})
  const [loading, setLoading] = useState(true)

  useEffect(() => {
    getPlayerById(id)
      .then(({ data }) => setPlayer(data))
      .catch((err) => console.error(err))
      .finally(() => setLoading(false))
  }, [id])

  if (loading) return <Loading />

  return (
    <>
      <Link to="/players">Back to Players</Link>
      <h1>{player.name}</h1>
      <p>
        {player.position} for the {player.teams.name}
      </p>
      {user && (
        <p>
          <Link to={`/players/${id}/edit`}>Edit Player</Link>
        </p>
      )}
    </>
  )
}

export default Player
