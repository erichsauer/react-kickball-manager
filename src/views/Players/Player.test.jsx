import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import Player from './Player'

it('should render a playerâ€™s details', async () => {
  const { container } = render(
    <MemoryRouter>
      <Player match={{ params: { id: 1 } }} />
    </MemoryRouter>
  )
  screen.getByLabelText(/loading/)
  await screen.findByText(/Ben E. Jetts/)

  expect(container).toMatchSnapshot()
})
