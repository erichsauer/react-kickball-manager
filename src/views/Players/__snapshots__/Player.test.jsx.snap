// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a playerâ€™s details 1`] = `
<div>
  <a
    href="/players"
  >
    Back to Players
  </a>
  <h1>
    Ben E. Jetts
  </h1>
  <p>
    Pitcher
     for the 
    Bridge City Sneakers
  </p>
</div>
`;
